nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-core/src/test/java/org/nuxeo/ecm/platform/audit/io/LogEntryCSVWriterTest.java

```diff
@@ -0,0 +1,136 @@
+        List<LogEntry> entries = (List<LogEntry>) pp.getCurrentPage();
+        if (entries.isEmpty()) {
+            // if there is nothing to test, the test would always succeed.
+            fail();
+        }
+        // No fetch param in the context, retrieve all default properties
+        RenderingContext renderingCtx = RenderingContext.CtxBuilder.get();
+        for (LogEntry entry : entries) {
+            CSVAssert csv = csvAssert(entry, renderingCtx);
+            csv.has("id").toString();
+            csv.has("category").isEquals("eventDocumentCategory");
+            csv.has("principalName").isEquals("Administrator"); // NOSONAR
+            csv.has("comment").isEquals("");
+            csv.has("docLifeCycle").isEquals("project"); // NOSONAR
+            csv.has("docPath").isEquals("/myFile"); // NOSONAR
+            csv.has("docType").isEquals("File"); // NOSONAR
+            csv.has("docUUID").isEquals(file.getId());
+            csv.has("eventId").isEquals(DocumentEventTypes.DOCUMENT_CREATED);
+            csv.has("repositoryId").isEquals("test");
+            csv.has("eventDate");
+            csv.has("logDate");
+        }
+    }
+
+    @Test
+    public void testCustomProperties() throws IOException {
+        DocumentModel file = session.createDocumentModel("/", "myFile", "File");
+        file = session.createDocument(file);
+        // LogEntry are event driven. They are created asynchronously
+        txFeature.nextTransaction();
+        PageProvider<?> pp = pps.getPageProvider("DOCUMENT_HISTORY_PROVIDER", null, Long.valueOf(20), Long.valueOf(0),
+                new HashMap<>(), file);
+        @SuppressWarnings("unchecked")
+        List<LogEntry> entries = (List<LogEntry>) pp.getCurrentPage();
+        if (entries.isEmpty()) {
+            // if there is nothing to test, the test would always succeed.
+            fail();
+        }
+        // Fetching only specific properties via context param
+        RenderingContext renderingCtx = //
+                RenderingContext.CtxBuilder.fetch(LogEntryCSVWriter.ENTITY_TYPE, "principalName", "docLifeCycle",
+                        "docPath", "docType").get();
+        for (LogEntry entry : entries) {
+            CSVAssert csv = csvAssert(entry, renderingCtx);
+            csv.has("principalName").isEquals("Administrator");
+            csv.has("docLifeCycle").isEquals("project");
+            csv.has("docPath").isEquals("/myFile");
+            csv.has("docType").isEquals("File");
+            // Default properties that are not requested and shouldn't be there
+            List<String> unwantedProps = List.of("category", "comment", "eventId", "eventDate", "logDate");
+            for (String property : unwantedProps) {
+                // expected to catch
+                try {
+                    csv.has(property);
+                    fail();
+                } catch (AssertionError e) {
+                    // Assert#fail would give an NPE here
```

Comment:
You can replace your comment by:
```
assertNotNull(e.getMessage());
```

Furthermore, you can assert the whole message as the exception message is:
```
"no field " + path
```
where path is `property` here.